assignments/this.js

```diff
@@ -1,26 +1,67 @@
 /* The for principles of "this";
 * in your own words. explain the four principle for the "this" keyword below.
 *
-* 1. 
-* 2. 
-* 3. 
-* 4. 
+* 1. When in the global scope, the value of “this” will be the window/console Object
+* 2. Whenever a function is called by a preceding dot, the object before that dot is this.
+* 3. Whenever a constructor function is used, this refers to the specific instance of the object that is created and returned by the constructor function.
+* 4. Whenever JavaScript’s call or apply method is used, this is explicitly defined.
 *
 * write out a code example of each explanation above
 */
 
 // Principle 1
-
+function saysName(name){
+    console.log(this);
+    return name;
+}
+saysName('Carlos Santana');*/
 // code example for Window Binding
 
 // Principle 2
+const myObj = {
+    greeting: 'Hello',
+    sayHello: function(name) {
```

Comment:
```suggestion
    sayHello: function(name) {
```
I would have made name a property of myObj and accessed it by this.name, rather than as an argument